Client/Frontend/Browser/Tabs/RecentlyClosedTabs/RecentlyClosedTabsView.swift

```diff
@@ -0,0 +1,155 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import Shared
+import BraveShared
+import Data
+import CoreData
+import BraveUI
+import BraveCore
+
+struct RecentlyClosedTabsView: View {
+  @Environment(\.presentationMode) @Binding private var presentationMode
+
+  @State var recentlyClosedTabs: [RecentlyClosed] = []
+  @State private var recentlyClosedLoading = true
+  
+  @State private var showClearDataPrompt: Bool = false
+  var onDismiss: ((Bool) -> Void)?
+  var onRecentlyClosedSelected: ((RecentlyClosed) -> Void)?
+  
+  private let tabManager: TabManager?
+
+  private var clearAllDataButton: some View {
+    Button(Strings.RecentlyClosed.recentlyClosedClearActionTitle, action: {
+      showClearDataPrompt = true
+    })
+    .accessibility(label: Text(Strings.RecentlyClosed.recentlyClosedClearActionConfirmation))
+    .foregroundColor(Color(.braveBlurpleTint))
+    .actionSheet(isPresented: $showClearDataPrompt) {
+      .init(title: Text(Strings.RecentlyClosed.recentlyClosedClearActionConfirmation),
+        buttons: [
+          .destructive(Text(Strings.RecentlyClosed.recentlyClosedClearActionTitle), action: {
+            RecentlyClosed.removeAll()
+            dismissView(cleared: true)
+          }),
+          .cancel()
+        ]
+      )
+    }
+  }
+  
+  private var doneButton: some View {
+    Button(Strings.done, action: {
+      dismissView()
+    })
+    .foregroundColor(Color(.braveBlurpleTint))
+  }
+  
+  private var websitesList: some View {
+    List {
+      Section {
+        ForEach(recentlyClosedTabs, id: \.url) { recentlyClosed in
+          Button(action: {
+            dismissView()
+            onRecentlyClosedSelected?(recentlyClosed)
+          }) {
+            HStack {
+              FaviconImage(url: recentlyClosed.url)  
+              VStack(alignment: .leading) {
+                Text(recentlyClosed.title ?? "")
+                  .font(.footnote)
+                  .fontWeight(.semibold)
+                  .foregroundColor(Color(.bravePrimary))
+                  .lineLimit(1)
```

Comment:
The reason we use one liner is some links might be too long to display, this is what we do everywhere.

Otherwise it can look like this.

![IMG_1654 2](https://user-images.githubusercontent.com/6643505/212769891-6225adaa-8663-4972-b2b5-4a324a65dc38.png)



And it works very well with dynamic fonts

![export1 2 2 1 2 3](https://user-images.githubusercontent.com/6643505/212769725-99f8e5da-13d9-49aa-9da4-7fb532b1c9c8.png)